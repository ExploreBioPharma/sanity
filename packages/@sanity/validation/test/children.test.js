const {Rule} = require('../src')

describe('child rules', () => {
  test('all() rules', () => {
    const rule = Rule.string().all([
      Rule.string().regex(/^[A-Z]/).error('Must start with an uppercase character'),
      Rule.string().regex(/[a-z]+/).error('Must follow with lowercase characters')
    ])

    expect(rule.validate('moop')).toMatchSnapshot('all() rules - failure')
    expect(rule.validate('Sanity')).toMatchSnapshot('all() rules - match')
  })
})
