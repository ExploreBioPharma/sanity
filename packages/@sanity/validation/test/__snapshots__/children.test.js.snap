// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all() rules - match 1`] = `
Object {
  "errors": Array [],
  "warnings": Array [],
}
`;

exports[`all() rules - match 2`] = `
Object {
  "errors": Array [],
  "warnings": Array [],
}
`;

exports[`all() rules - multiple failures, common error 1`] = `
Object {
  "errors": Array [
    [ValidationError: Needs to be a capital letter followed by at least 4 lowercase characters],
  ],
  "warnings": Array [],
}
`;

exports[`all() rules - multiple failures, custom messages 1`] = `
Object {
  "errors": Array [
    [ValidationError: [Must start with an uppercase character - AND - 5 chars or more]],
  ],
  "warnings": Array [],
}
`;

exports[`all() rules - single failure, common error 1`] = `
Object {
  "errors": Array [
    [ValidationError: Needs to start with a capital letter and then follow with lowercase characters],
  ],
  "warnings": Array [],
}
`;

exports[`all() rules - single failure, custom message 1`] = `
Object {
  "errors": Array [
    [ValidationError: Must start with an uppercase character],
  ],
  "warnings": Array [],
}
`;

exports[`either() rules - all fail, common error 1`] = `
Object {
  "errors": Array [
    [ValidationError: Needs to start with a capital letter, unless it's an iProduct],
  ],
  "warnings": Array [],
}
`;

exports[`either() rules - all match 1`] = `
Object {
  "errors": Array [],
  "warnings": Array [],
}
`;

exports[`either() rules - match 1`] = `
Object {
  "errors": Array [
    [ValidationError: [Must be rgb(num, num, num) format - OR - Must be hex color with #-prefix]],
  ],
  "warnings": Array [],
}
`;

exports[`either() rules - no matches 1`] = `
Object {
  "errors": Array [
    [ValidationError: [Must start with a capital R - OR - Must end with "ed"]],
  ],
  "warnings": Array [],
}
`;

exports[`either() rules - single failure, custom message 1`] = `
Object {
  "errors": Array [
    [ValidationError: [Must be rgb(num, num, num) format - OR - Must be hex color with #-prefix]],
  ],
  "warnings": Array [],
}
`;
